### ISSUE TITLE
[REQUEST] AutoDL-iRSSi Plugins

### ISSUE THREAD

**Issue by geekism** [jfurrow/flood#447]
Anyway to have a plugin, or features for irssi's autodl script? Perhaps by using the pre-existing rutorrent plugins. Or can flood have a plugin system such as rutorrent has.

**Comment by jfurrow** [comment]
What features does this plugin provide?

**Comment by randomnonsense** [comment]
I don't understand what @geekism means either. Irssi pluging downloads new torrents via IRC as a bot announces them, e.g. on a private tracker. How is flood supposed to do anything with that? You can have both flood and your irssi using a watch folder. Flood will display such torrents assigning them a tag.

**Comment by jvacek** [comment]
As @randomnonsense described, autodl-irssi snatches torrents out of IRC.
Rutorrent had a plugin that made it super easy to work with autodl-irssi for configuration, logging, etc. within the web interface
https://www.rapidseedbox.com/kb/use-rutorrents-autodl-irssi-plugin
https://www.rapidseedbox.com/wp-content/uploads/filter-general.png
https://www.rapidseedbox.com/wp-content/uploads/IRC-CHANNELS.png

**Comment by Rick7C2** [comment]
I just found out about flood. It looks awesome! But until it has native support for autodl-irssi I have to use rutorrent.

I will be watching this issue. :)

**Comment by hollanbm** [comment]
> I just found out about flood. It looks awesome! But until it has native support for autodl-irssi I have to use rutorrent.

>I will be watching this issue. :)

Same here. Plus ruTorrent isn't currently compatible with ubuntu 18.04 due to requiring php 7.0. I'd love to upgrade to something more modern, but without autodl support, I won't be making the leap :(

**Comment by noraj** [comment]
This is not AutoDL-iRSSi but the flood Feed & Downlaod seems to do the same?

You can watch some RSS feed and make automatic downloads based on regex rules.

![](http://i.imgur.com/N3d0WZR.png)

Is there something different?

**Comment by qcsunny** [comment]
"autodl-irssi" can download torrents faster,easier and more efficient  than "rss"

**Comment by SavageCore** [comment]
To clarify, autodl-irssi gets near real time uploads via IRC whereas RSS is slowly polled, useless for racing.

**Comment by noraj** [comment]
Yeah that's true, I forgot about the IRC capacity, I used it a long time ago.

But before implementing an autodl-irssi like plugin, flood needs a [plugin API](https://github.com/jfurrow/flood/issues/647)

**Comment by robertparkerx** [comment]
This would be really nice to have :)
